target/linux/ar71xx/files/arch/mips/ath79/mach-rbspi.c

```diff
@@ -932,12 +933,19 @@ static void __init rbmap_setup(void)
 	/* GMAC1 is HW MAC, WLAN0 MAC is HW MAC + 2 */
 	rbspi_network_setup(flags, 0, 2, 0);
 
-	if (flags & RBSPI_HAS_POE)
-		gpio_request_one(RBMAP_GPIO_POE_POWER,
-				GPIOF_OUT_INIT_HIGH | GPIOF_EXPORT_DIR_FIXED,
-				"POE power");
+	/* USB power GPIO is inverted, set GPIOF_ACTIVE_LOW for consistency */
```

Comment:
you removed POE power, that board doesn't have POE passthrough power control?

Reply:
Yes, it does.
I have added GPIO switch to it.
That works without a issue.

Reply:
your last commit has removed the GPIO request for POE power. It can't work anymore.

Reply:
It can, there is no need for GPIO request If it is not needed for it to be on by default.
GPIO switch does its job.

Reply:
> It can, there is no need for GPIO request If it is not needed for it to be on by default.
> GPIO switch does its job.

But you probably don't want to allow user to change the GPIO direction?

Reply:
That, exactly. gpio_request_one() serves more purpose than what can be done in the userspace scripts.

Reply:
As for the "on by default" it was my understanding that this is the default behavior in RouterOS, and a desirable one, but maybe I was wrong... It can simply be changed with the  GPIOF_OUT_INIT_LOW flag if necessary.

Reply:
I was not thinking about users changing direction.
Will use gpio_request_one() for POE instead of GPIO switch.

No, RouterOS does not enable POE by default.
Otherwise, you will fry any device like a notebook when trying to configure the device first time since you can only do that through ETH2 port.

Reply:
i'm not sure how you could fry anything with POE, especially considering it's passthrough, but I stand corrected for the RouterOS default :)

Reply:
Changed to gpio_request_one()

Reply:
Well you can easily fry networking cards or whole devices.
Since device passes the same voltage as received for powering via DC jack or POE IN on first port.
So if you use 24V DC power brick or passive 24V POE that is commonly used and pass that to notebook or non POE powered routers and you can easily damage or destroy them

Reply:
> i'm not sure how you could fry anything with POE, especially considering it's passthrough, but I stand corrected for the RouterOS default :)

Passive PoE doesn't do any negotiation, it just put voltage to the cable.

Reply:
Well sure but I still hope it does it right by putting the same voltage on each differential pair. In which case it's essentially harmless. That's how true POE ensures backward safety with non-POE compliant hardware...

Reply:
Passive POE does not do that.
It just puts voltage between 2 pairs.
Usually 4,5 is positive and 7,8 negative

Reply:
That's perfectly fine and harmless. POE works out autodetection by putting a voltage (2-10V) on the pairs. It's perfectly ok to have a voltage _between_ pairs, as long as there's no DC voltage between each side of a single pair.
If a device is damaged in this context, clearly the hardware does not follow the ethernet standard, which requires transformer coupling... Or there's something else I'm not understanding here.

Reply:
Replying to myself, I see where the problem is: the passive POE voltage might be too high for the other side to cope with if it isn't compliant. Point taken :)